### ISSUE TITLE
recommend flag of available package prioritize another package even though is disfavored

### ISSUE THREAD

**Issue by jsilhan** [openSUSE/libsolv#139]
HI,
is it right to consider `recommends` flag of non installed package?
```
repo system 0 testtags <inline>
repo available 0 testtags <inline>
#>=Pkg: B 1 0 noarch
#>=Rec: C
#>=Prv: somereq
#>=Obs: somereq
#>=Pkg: C 1 0 noarch
#>=Prv: somereq
#>=Pkg: somereq 1 0 noarch
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job install provides somereq
job disfavor name A
job disfavor name C
result transaction,problems <inline>
#>install C-1-0.noarch@available
```
expected result: `B`

When I remove the `Rec: C`, the `B` package gets installed.

**Comment by mlschroe** [comment]
That's actually not preference, but minimization. The solver thinks that installing B will also lead to C (because of the recommends), so it reverts to just installing C.

Seems that's one place in the code where I forgot to check the disfavors. Fixing...

**Comment by mlschroe** [comment]
Fixed with commit 8c94293991ad8be57b99f8c413a3596b62649fa6

**Comment by jsilhan** [comment]
Thanks, it works
